node_modules/gh-series/lib/internal/patterns/cambridge/listener.js

```diff
@@ -0,0 +1,170 @@
+            // we have easy access to the `event.organisers` property
+            .map(function(event) {
+                return event.toJSON();
+            })
+
+            // Iterate over each event and keep track of the pattern, location and organisers
+            .each(function(event) {
+                fullPattern.add(event);
+                locations.push(event.location);
+                organisers.push(event.organisers);
+            });
+
+        // Only retain the unique locations
+        locations = _.chain(locations).uniq().compact().value().sort();
+
+        // Only retain the unique names of the organisers
+        organisers = _.chain(organisers)
+            .flatten()
+            .map(function(organiser) {
+                if (_.isObject(organiser)) {
+                    return organiser.displayName;
+                } else {
+                    return organiser;
+                }
+            })
+            .uniq()
+            .compact()
+            .value()
+            .sort();
+
+        var metadata = {
+            'pattern': fullPattern.toString(),
+            'locations': locations,
+            'organisers': organisers
+        };
+
+        // Update the serie metadata
+        SeriesDAO.updateSerie(serie, {'metadata': metadata}, function(err) {
+            if (err) {
+                log().error({
+                    'err': err,
+                    'serie': serie.id
+                }, 'Unable to persist a pattern for a serie');
+            }
+
+            // Indicate that the asynchronous metadata-generation operation has completed
+            Listener.emit('postGeneration', serie);
+        });
+    });
+};
+
+/**
+ * Called when events are added or removed from an event series. When the serie
+ * resides on a `timetable` application, the metadata for it will be (re)generated
+ *
+ * @param  {Context}    ctx         Standard context containing the current user and the current app
+ * @param  {Serie}      serie       The serie to/from which events were added/removed
+ * @api private
+ */
+var onEventSerieChange = function(ctx, serie) {
+    if (serie.App.type === 'timetable') {
+
+        // Indicate that we're about to generate some serie metadata allowing the tests
+        // to pause til the asynchronous operation finishes
+        Listener.emit('preGeneration');
+
+        // Generate the metadata for this serie
+        generateMetadata(serie);
+    }
+};
+
+SeriesAPI.on(SeriesConstants.events.ADDED_EVENTS, onEventSerieChange);
+SeriesAPI.on(SeriesConstants.events.DELETED_EVENTS, onEventSerieChange);
+
+EventsAPI.on(EventsConstants.events.UPDATED_EVENT, function(ctx, event, updatedEvent) {
+    // Only re-generate patterns if the event belongs to the timetable app
+    // and the start and/or end date of the event changed
+    // TODO: Organisers
+    if (updatedEvent.App.type === 'timetable' && (
+         (event.start !== updatedEvent.start || event.end !== updatedEvent.end)) ||
```

Comment:
Yes, but we don't have the necessary CRUD functionality to add/remove/update organisers yet